Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/CalibrationAPI.cs

```diff
@@ -0,0 +1,389 @@
+            var cameraTransform = Matrix4x4.TRS(data.headsetData.position, data.headsetData.rotation, Vector3.one);
+            var inverseCameraTransform = cameraTransform.inverse;
+            foreach (var markerPair in data.markers)
+            {
+                var arUcoCorners = markerPair.arucoMarkerCorners;
+
+                var cornersRelativeToCamera = new MarkerCorners();
+                cornersRelativeToCamera.topLeft = inverseCameraTransform.MultiplyPoint(arUcoCorners.topLeft);
+                cornersRelativeToCamera.topRight = inverseCameraTransform.MultiplyPoint(arUcoCorners.topRight);
+                cornersRelativeToCamera.bottomRight = inverseCameraTransform.MultiplyPoint(arUcoCorners.bottomRight);
+                cornersRelativeToCamera.bottomLeft = inverseCameraTransform.MultiplyPoint(arUcoCorners.bottomLeft);
+                var orientationTransform = inverseCameraTransform * Matrix4x4.TRS(Vector3.zero, arUcoCorners.orientation, Vector3.one);
+                cornersRelativeToCamera.orientation = Quaternion.LookRotation(orientationTransform.GetColumn(2), orientationTransform.GetColumn(1));
+                markersRelativeToCamera.Add(cornersRelativeToCamera);
+            }
+            return markersRelativeToCamera;
+        }
+
+        public List<CalculatedCameraExtrinsics> CalculateIndividualArUcoExtrinsics(CameraIntrinsics intrinsics, int numExtrinsics)
+        {
+            int intrinsicsSize = 12;
+            float[] inputIntrinsics = new float[intrinsicsSize];
+            inputIntrinsics[0] = intrinsics.FocalLength.x;
+            inputIntrinsics[1] = intrinsics.FocalLength.y;
+            inputIntrinsics[2] = intrinsics.PrincipalPoint.x;
+            inputIntrinsics[3] = intrinsics.PrincipalPoint.y;
+            inputIntrinsics[4] = intrinsics.RadialDistortion.x;
+            inputIntrinsics[5] = intrinsics.RadialDistortion.y;
+            inputIntrinsics[6] = intrinsics.RadialDistortion.z;
+            inputIntrinsics[7] = intrinsics.TangentialDistortion.x;
+            inputIntrinsics[8] = intrinsics.TangentialDistortion.y;
+            inputIntrinsics[9] = intrinsics.ImageWidth;
+            inputIntrinsics[10] = intrinsics.ImageHeight;
+            inputIntrinsics[11] = 0.0f; // reprojection error, unused
+
+            int sizeExtrinsics = 7;
+            float[] extrinsics = new float[numExtrinsics * sizeExtrinsics];
+            if (!ProcessIndividualArUcoExtrinsicsNative(inputIntrinsics, extrinsics, numExtrinsics))
+            {
+                PrintLastError();
+                return null;
+            }
+            else
+            {
+                List<CalculatedCameraExtrinsics> output = new List<CalculatedCameraExtrinsics>();
+                for (int i = 0; i < numExtrinsics; i++)
+                {
+                    int offset = i * sizeExtrinsics;
+                    bool succeeded = (extrinsics[offset] > 0.01f);
```

Comment:
>(extrinsics[offset] > 0.01f) [](start = 37, length = 28)

This may be answered in some docs though I'm not familiar - what's the special reason why 0.01f is chosen here as "success"?

Reply:
on this should just be zero